iis-wf/iis-wf-import/src/test/java/eu/dnetlib/iis/wf/importer/infospace/truncator/DocumentMetadataAvroTruncatorTest.java

```diff
@@ -0,0 +1,156 @@
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(abstractTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateAbstractWhenItIsNotNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+        source.setAbstract$("abstract");
+        when(abstractTruncator.apply("abstract")).thenReturn("truncated");
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals("truncated", result.getAbstract$());
+    }
+
+    @Test
+    public void shouldNotTruncateTitleWhenItIsNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(titleTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateTitleWhenItIsNotNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+        source.setTitle("title");
+        when(titleTruncator.apply("title")).thenReturn("truncated");
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals("truncated", result.getTitle());
+    }
+
+    @Test
+    public void shouldNotTruncateAuthorsWhenItIsNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(authorsTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateAuthorsWhenItIsNotNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+        List<Author> authors = Collections.singletonList(mock(Author.class));
+        source.setAuthors(authors);
+        List<Author> truncated = Collections.singletonList(mock(Author.class));
+        when(authorsTruncator.apply(authors)).thenReturn(truncated);
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(truncated, result.getAuthors());
+    }
+
+    @Test
+    public void shouldNotTruncateKeywordsWhenItIsNull() {
+        // given
+        DocumentMetadata source = createDocumentMetadata();
+
+        // when
+        DocumentMetadata result = documentMetadataAvroTruncator.truncate(source);
+
+        // then
+        assertEquals(source, result);
+        verify(keywordsTruncator, never()).apply(any());
+    }
+
+    @Test
+    public void shouldTruncateKeywordsWhenItIsNotNull() {
```

Comment:
Same here: I guess we could also add a test verifying whether the number of keywords is properly truncated.

Reply:
And the same answer applies here :)